src/laser/Laser.cpp

```diff
@@ -35,70 +55,671 @@ Laser::InitData (const amrex::BoxArray& slice_ba,
+
+        // Multiply by appropriate factors in Fourier space
+        amrex::Real dkx = 2.*MathConst::pi/geom.ProbLength(0);
+        amrex::Real dky = 2.*MathConst::pi/geom.ProbLength(1);
+        // acoeff_imag is supposed to be a nx*ny array.
+        // For the sake of simplicity, we evaluate it on-axis only.
+        const Complex acoeff =
+            step == 0 ? 6._rt/(c*dt*dz) - I * 4._rt * ( k0 + djn ) / (c*dt) :
+             3._rt/(c*dt*dz) + 2._rt/(c*c*dt*dt) - I * 2._rt * ( k0 + djn ) / (c*dt);
+        amrex::ParallelFor(
+            bx,
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                // divide rhs_fourier by -(k^2+a)
+                amrex::Real kx = (i<imid) ? dkx*i : dkx*(i-Nx);
+                amrex::Real ky = (j<jmid) ? dky*j : dky*(j-Ny);
+                const Complex inv_k2a = abs(kx*kx + ky*ky + acoeff) > 0. ?
+                    1._rt/(kx*kx + ky*ky + acoeff) : 0.;
+                rhs_fourier_arr(i,j,k,0) *= -inv_k2a;
+            });
+
+        // Transform rhs to Fourier space to get solution in sol
+#ifdef AMREX_USE_CUDA
+        cudaStream_t stream_bkw = amrex::Gpu::Device::cudaStream();
+        cufftSetStream ( m_plan_bkw, stream_bkw);
+        result = LaserFFT::VendorExecute(
+            m_plan_bkw,
+            reinterpret_cast<LaserFFT::cufftComplex*>( m_rhs_fourier.dataPtr() ),
+            reinterpret_cast<LaserFFT::cufftComplex*>( m_sol.dataPtr() ),
+            CUFFT_INVERSE);
+        if ( result != CUFFT_SUCCESS ) {
+            amrex::Print() << " forward transform using cufftExec failed ! Error: " <<
+                CuFFTUtils::cufftErrorToString(result) << "\n";
+        }
+#elif defined(AMREX_USE_HIP)
+        amrex::Abort("Not implemented");
+#else
+        LaserFFT::VendorExecute( m_plan_bkw );
+#endif
+
+        // Normalize and store solution in np1j00[0]. Guard cells are filled with 0s.
+        amrex::Box grown_bx = bx;
+        grown_bx.grow(m_slices_nguards);
+        const amrex::Real inv_numPts = 1./bx.numPts();
+        amrex::ParallelFor(
+            grown_bx,
+            [=] AMREX_GPU_DEVICE(int i, int j, int) noexcept {
+                if (i>=imin && i<=imax && j>=jmin && j<=jmax) {
+                    np1j00_arr(i,j,0) = sol_arr(i,j,0).real() * inv_numPts;
+                    np1j00_arr(i,j,1) = sol_arr(i,j,0).imag() * inv_numPts;
+                } else {
+                    np1j00_arr(i,j,0) = 0._rt;
+                    np1j00_arr(i,j,1) = 0._rt;
+                }
+            });
+    }
+}
+
+void
+Laser::InitLaserSlice (const amrex::Geometry& geom, const int islice)
+{
+    if (!m_use_laser) return;
+
+    HIPACE_PROFILE("Laser::InitLaserSlice()");
+
+    using namespace amrex::literals;
+    using Complex = amrex::GpuComplex<amrex::Real>;
+
+    // Basic laser parameters and constants
+    Complex I(0,1);
+    const amrex::Real c = get_phys_const().c;
     const int dcomp = 0; // NOTE, this may change when we use slices with Comps
```

Comment:
```suggestion
    const int dcomp = 0;
```

Reply:
Somehow, this comment is still accurate. But indeed, it would apply to many places, so let's just remove it.